pkg/steps/source.go

```diff
@@ -37,22 +38,29 @@ const (
 
 	ProwJobIdLabel = "prow.k8s.io/id"
 
-	gopath = "/go"
+	gopath     = "/go"
+	sshKeyPath = "/sshprivatekey"
 )
 
 var (
 	JobSpecAnnotation = fmt.Sprintf("%s/%s", CiAnnotationPrefix, "job-spec")
 )
 
-func sourceDockerfile(fromTag api.PipelineImageStreamTagReference, workingDir string) string {
-	return fmt.Sprintf(`
-FROM %s:%s
-ADD ./app.binary /clonerefs
-RUN umask 0002 && /clonerefs && find %s/src -type d -not -perm -0775 | xargs -r chmod g+xw
-WORKDIR %s/
-ENV GOPATH=%s
-RUN git submodule update --init
-`, api.PipelineImageStream, fromTag, gopath, workingDir, gopath)
+func sourceDockerfile(fromTag api.PipelineImageStreamTagReference, workingDir string, sshSecret *coreapi.Secret) string {
+	var dockerCommands []string
+	dockerCommands = append(dockerCommands, fmt.Sprintf("FROM %s:%s", api.PipelineImageStream, fromTag))
+	dockerCommands = append(dockerCommands, "ADD ./app.binary /clonerefs")
+
+	if sshSecret != nil {
+		dockerCommands = append(dockerCommands, fmt.Sprintf("RUN echo %s > %s", string(sshSecret.Data[coreapi.SSHAuthPrivateKey]), sshKeyPath))
```

Comment:
I'll do a full review later, but this caught my eye while I was skimming the changes: we don't want to record the private key in plain text in the builds/logs. See:

https://docs.okd.io/latest/dev_guide/builds/build_inputs.html#using-secrets-during-build

Reply:
Oops, yea you are right. So this will become more complicated. Btw, the link you commend above is not for this case. I already add the source secret for builds, but that Dockerfile is different, because we need to inject that ssh key to the clonerefs container and let `cloneref` to use it. I guess we will need to inject a shared Volume in that build, etc....

Reply:
No, we should use the secret injection mechanism linked by Bruno to inject the secrets into this build, too. The `clonerefs` options just need to reference it.

Reply:
Yep, I will address this.